ContosoCrafts.WebSite/Models/User.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+
+
+namespace ContosoCrafts.WebSite.Models
+{
+	public class User
+	{
+        public string UserId { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public DateTime JoinedAt { get; set; }
+        public string Username { get; set; }
+        public string PasswordHash { get; set; }
+        public string ProfilePictureUrl { get; set; }
+
+
+        public override string ToString() => JsonSerializer.Serialize<User>(this);
+
+        // Method to hash the password using SHA256
+        public void HashPassword(string password)
+        {
+            using (SHA256 sha256Hash = SHA256.Create())
+            {
+                // Compute hash from the password string
+                byte[] bytes = sha256Hash.ComputeHash(Encoding.UTF8.GetBytes(password));
+
+                // Convert byte array to a string
+                StringBuilder sb = new StringBuilder();
+                foreach (byte b in bytes)
+                {
+                    sb.Append(b.ToString("x2"));
+                }
+                PasswordHash = sb.ToString();
+            }
+        }
+
+        // Method to verify if the provided password matches the stored hash
```

Comment:
Where does the password and user data get stored?

Reply:
I was thinking it would be written to a file.